### PR TITLE
Add support for products with services and attachments

### PR DESCRIPTION
**Pull request by victorhmp**
#### What is the purpose of this pull request?
Add support for products that have `attachments` or `bundle items`, along with support for the new assembly options API.

#### What problem is this solving?
Attachments and bundle items were not supported and would be ignored by the `Product` component.

#### How should this be manually tested?
`cd react && yarn test`

#### Screenshots or example usage

screen shot 2019-02-10 at 1 06 41 pm

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: messages/context.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  "warnings.payment.approval": "Warning about payment taking up to three days to be approved",
+  "warnings.payment.bankInvoice.approval": "Warning about payment approvement for bank invoices, noting that if th bank invoice is not due, order will be cancelled",
+  "warnings.payment.bankInvoice.value.duedate": "Warning to show the value and due date of bank invoice payment",
+  "warnings.payment.bankInvoice.value": "Warning to show the value (and no due date) of bank invoice payment",
+  "warnings.delivery.time": "Warning the user that delivery period starts from the moment payment is confirmed",
+  "warnings.delivery.tracking": "Warning the user that when his/her order is on its way, a tracking code will be sent to his/her email",
+  "warnings.pickup.time": "Warning the user that the store pickup period starts from the moment payment is confirmed",
+  "warnings.order.split": "Warning the user that his/her purchase was split into multiple orders as some of the items were sold by partners and that does not affect shipping estimates",
+  "summary.shipping": "Tittle for the section that shows a summary of items to be delivered",
```
Comment (lbebber) [inline messages/context.json]:
```suggestion
  "summary.shipping": "Title for the section that shows a summary of items to be delivered",
```

File: messages/context.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  "summary.pickup": "Tittle for the section that shows a summary of items to be picked-up",
```
Comment (lbebber) [inline messages/context.json]:
```suggestion
  "summary.pickup": "Title for the section that shows a summary of items to be picked-up",
```

File: messages/context.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
+{
+  "country.BRA": "Brazil",
+  "header.thanks": "Message to thank the user for placing an order",
+  "header.email": "Disclaimer about the user receiving an email with the information about placed order",
+  "header.email.button": "A button in the header to resend order information email",
+  "header.print.button": "Button in the header to print the page",
+  "header.newpurchase.button": "Button in the header to start a new order",
+  "header.bankinvoice.header": "Tittle for a section that shows information on generated bank invoice",
```
Comment (lbebber) [inline messages/context.json]:
```suggestion
  "header.bankinvoice.header": "Title for a section that shows information on generated bank invoice",
```

Comment (lbebber) [reply messages/context.json]: "Tittle" é o pinguinho do i

File: react/components/Shipping/ShippingHeader.tsx [hunk 4]
```diff
@@ -39,7 +41,17 @@ const ShippingHeader: FunctionComponent<Props & InjectedIntlProps> = ({
         <br />
         <small className="c-muted-2 t-small">{shippingData.selectedSla}</small>
       </p>
-      <Address address={shippingData.address} />
+      {giftRegistry &&
+      giftRegistry.addressId === shippingData.address.addressId ? (
```
Comment (lbebber) [inline react/components/Shipping/ShippingHeader.tsx]: i see you are checking if `giftRegistry` exists. is `shippingData...` guaranteed to exist by this point?

File: react/components/ProductList/Attachment.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                    className="c-action-primary ml5"
+                    onClick={() => setIsOpen(!isOpen)}
+                  >
+                    {isOpen ? <IconCaretUp /> : <IconCaretDown />}
+                  </div>
+                )}
+              </div>
+            </div>
+            {bundleItem.attachments && bundleItem.attachments.length > 0 && (
+              <div hidden={!isOpen}>
```
Comment (lbebber) [inline react/components/ProductList/Attachment.tsx]: not sure about this `hidden`, but w/e

File: messages/context.json (hunk 1 above)
Comment (victorhmp) [reply messages/context.json]: Fixed

File: messages/context.json (hunk 2 above)
Comment (victorhmp) [reply messages/context.json]: Fixed

File: messages/context.json (hunk 3 above)
Comment (victorhmp) [reply messages/context.json]: Fixed

File: react/components/Shipping/ShippingHeader.tsx (hunk 4 above)
Comment (victorhmp) [reply react/components/Shipping/ShippingHeader.tsx]: Yep, some of them could be `null`, but they are not `undefined`, so nothing should break.
My interface definition for `Parcel`, which is the type of `shippingData`:

```
interface Parcel {
  address: Address
  price: number
  pickupFriendlyName: string | null
  seller: string
  items: OrderItem[]
  selectedSla: string
  selectedSlaObj: ShippingSLA
  shippingEstimate: string | null
  shippingEstimateDate: string | null
  deliveryWindow: DeliveryWindow | null
  deliveryChannel: string
  selectedSlaType: string
}
```

File: react/components/ProductList/Attachment.tsx (hunk 5 above)
Comment (victorhmp) [reply react/components/ProductList/Attachment.tsx]: I didn't understand the problem hahahah

File: react/graphql/getOrderGroup.graphql [hunk 6]
```diff
@@ -30,6 +30,28 @@ query getOrderGroup($orderGroup: String) {
           listPrice
           isGift
           quantity
+          attachments {
+            content {
+              text
```
Comment (klzns) [inline react/graphql/getOrderGroup.graphql]: Attachment content have arbitrary keys and values

File: react/graphql/getOrderGroup.graphql [hunk 7]
```diff
@@ -30,6 +30,28 @@ query getOrderGroup($orderGroup: String) {
...
+              periodo
+            }
+            name
+          }
+          bundleItems {
+            id
+            attachments {
+              content {
+                text
```
Comment (klzns) [inline react/graphql/getOrderGroup.graphql]: Same

File: react/graphql/getOrderGroup.graphql (hunk 6 above)
Comment (klzns) [reply react/graphql/getOrderGroup.graphql]: See https://github.com/vtex/checkout-confirmation-ui/blob/master/src/mock/mockAttachments.json#L55

File: react/components/ProductList/Attachment.tsx (hunk 5 above)
Comment (victorhmp) [reply react/components/ProductList/Attachment.tsx]: I thought it was necessary to handle bundle items that did not have attachments to them. Something like this:

```
items: [
              {
                id: '22',
                skuName: 'Sku serviço',
                name: 'Produto serviço Sku serviço',
                price: 1299,
                listPrice: 1500,
                isGift: false,
                quantity: 1,
                attachments: [],
                bundleItems: [
                  {
                    id: '2',
                    attachments: [],
                    name: '[TESTE QA]',
                    price: 500,
                    quantity: 1,
                    imageUrl: null,
                    measurementUnit: 'un',
                    unitMultiplier: 1,
                  },
                ],
                seller: '1',
                imageUrl:
                  'http://vtexgame1.vteximg.com.br/arquivos/ids/155359-55-55/Produto-com-servico.png?v=635812252385100000',
                detailUrl: '/produto-servico/p',
                measurementUnit: 'un',
                unitMultiplier: 1,
              },

```

Comment (victorhmp) [reply react/components/ProductList/Attachment.tsx]: I created this from `vtexgame1` store

File: react/components/ProductList/Attachment.tsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+                <p className="c-muted-1">
+                  {bundleItem.attachments[0].content.text}
```
[the remark this answers is not in the document]
Comment (victorhmp) [reply react/components/ProductList/Attachment.tsx]: Yep, I added a conditional to check if attachment is a message

File: react/components/ProductList/index.tsx [hunk 9]
```diff
@@ -1,15 +1,26 @@
...
-      <Product productInfo={product} key={product.id} />
-    ))}
-  </Fragment>
-)
+const ProductList: FunctionComponent<Props> = ({ products }) => {
+  return (
+    <Fragment>
+      {products.map((product: OrderItem) => (
+        <Fragment key={product.id}>
+          <Product productInfo={product} key={product.id} />
```
[the remark this answers is not in the document]
Comment (victorhmp) [reply react/components/ProductList/index.tsx]: Removed it

File: react/graphql/getOrderGroup.graphql (hunk 6 above)
Comment (victorhmp) [reply react/graphql/getOrderGroup.graphql]: That is so weird 😅
But I updated the query according to the adjustment make in PR #5 from `vtex/order-placed-graphql`

File: react/graphql/getOrderGroup.graphql (hunk 7 above)
Comment (victorhmp) [reply react/graphql/getOrderGroup.graphql]: Updated

File: react/components/ProductList/Attachment.tsx [hunk 10]
```diff
@@ -20,37 +20,64 @@ const ProductAttachment: FunctionComponent<Props & InjectedIntlProps> = ({
...
+          return (
+            <article className="bg-muted-5 pv3 ph5 br2 mv4" key={bundleItem.id}>
+              <div className="flex justify-between">
+                {bundleItem.imageUrl && (
+                  <ProductImage
+                    url={bundleItem.imageUrl}
+                    alt={bundleItem.name}
+                  />
                 )}
+                <p className="block c-on-base">{bundleItem.name}</p>
```
Comment (klzns) [inline react/components/ProductList/Attachment.tsx]: There is no class `block`

**Pull request merged**